tensorflow_federated/python/core/utils/execution_tracing.py

```diff
@@ -0,0 +1,313 @@
+
+  def propagate_trace_context_task_factory(self, loop, coro):
+    """No-op implementation of propagate_trace_context_task_factory."""
+    return asyncio.tasks.Task(coro, loop=loop)
+
+  def run_coroutine_threadsafe_in_ambient_trace_context(self, coro, loop):
+    """No-op implementation of run_coroutine_threadsafe_in_ambient_trace_context."""
+    return asyncio.run_coroutine_threadsafe(coro, loop)
+
+  def run_coroutine_threadsafe_in_task_trace_context(self, coro, loop):
+    """No-op implementation of run_coroutine_threadsafe_in_ambient_trace_context."""
+    return asyncio.run_coroutine_threadsafe(coro, loop)
+
+  def run_coroutine_in_ambient_trace_context(self, coro):
+    """No-op implementation of run_coroutine_in_ambient_trace_context."""
+    return coro
+
+  @classmethod
+  def _is_traceable(cls, method_name, regex, override=False):
+    if override:
+      return True
+    match = regex.search(method_name)
+    if match is None:
+      return False
+    return True
+
+  def _trace_method(self, fn, class_name, method_name, call_or_retr):
+    # rough workaround -- Python can't distinguish that fn is a method here
+    fn_sig = inspect.signature(fn)
+    has_self = 'self' in fn_sig.parameters
+
+    def _log_fn(*args, **kwargs):
+      nonlocal has_self
+
+      ctx = None
+      if has_self:
+        ctx = args[0]
+        args = args[1:]
+
+      prefix_str = self._prefix(ctx, class_name)
+      main_str = "{m} {c}".format(m=method_name, c=call_or_retr)
+      suffix_args = " ".join(self.format_object(a) for a in args)
+      suffix_kwargs = " ".join(self.format_object(v) for k, v in kwargs.items())
+      logging.debug("{p} {m} {sa} {sk}".format(
+          p=prefix_str, m=main_str, sa=suffix_args, sk=suffix_kwargs))
+
+    return _log_fn
+
+  def _prefix(self, ctx, type_):
+    indentation = '  ' * self._current_call_level
+    template = "{indentation}{type} {hash}"
+
+    if ctx is None:
+      return template.format(indentation=indentation, type=type_, hash="NO_ID")
+    return template.format(indentation=indentation, type=type_, hash=hash(ctx))
+
+  def _format_object(self, x):
+    strategy = self._formatting_strategies.get(type(x), self._default_strategy)
+    return strategy(x)
+
+  def _format_anon_tuple(self, tup):
+    names = tup._name_array
+    values = tup._element_array
+    fmt_strs = []
+    for n, v in zip(names, values):
+      fmt_strs.append("{}: {}".format(n, self._format_object(v)))
+    return "({})".format(", ".join(fmt_strs))
+
+  def _register_formatting_strategy(self, type_, formatting_strategy):
+    self._formatting_strategies[type_] = formatting_strategy
+
+  # Register all types here
+  def _make_formatting_strategies(self):
+    self.register_executors()
+    self.register_executor_values()
+    self.register_computations()
+    self.register_computation_types()
+    self.register_primitives()
+
+  def register_executors(self):
+    register = self._register_formatting_strategy
```

Comment:
Nit: I would suggest that we not alias the self describing name to something that is less self describing or we choose that the shorter name is self describing and rename the method to that. The alias seems unnecessary.